src/adaptors/mod.rs

```diff
@@ -606,107 +605,121 @@ impl<I, J, F> Iterator for MergeBy<I, J, F>
     }
 }
 
-#[derive(Clone, Debug)]
-pub struct CoalesceCore<I, T>
+/// An iterator adaptor that may join together adjacent elements.
+///
+/// See [`.coalesce()`](../trait.Itertools.html#method.coalesce) for more information.
+#[must_use = "iterator adaptors are lazy and do nothing unless consumed"]
+pub struct Coalesce<I, F, T>
```

Comment:
Unfortunately, adding an extra type parameter to `Coalesce` is a breaking change.

Reply:
That's a very good objection. I already tried to keep it on 2 parameters, but ran into problems since `Coalesce` has to know the result of `F` and I had problems specifying that.

Could I just rename `Coalesce` to `CoalesceBy` (keeping the type parameters) and define `type Coalesce<I, F> = CoalesceBy<I, F, I::Item>` (and the others alike)?

Reply:
Yep!